### PR TITLE
doc/rados: edit ops/control.rst (1 of x)

### PR DESCRIPTION
**Pull request by zdover23**
Edit doc/rados/operations/control.rst (1 of x).

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by zdover23** [comment]
See https://github.com/ceph/ceph/pull/53792 (edit ops/control.rst (2 of x)).

File: doc/rados/operations/control.rst [hunk 1]
```diff
@@ -60,7 +60,7 @@ Here the value of ``[id]`` can be found by consulting the output of ``ceph
 Authentication Subsystem
 ========================
 
-To add an OSD keyring for a specific OSD, run the following command:
+To add an OSD keyring to a specific OSD, run the following command:
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:63]: I think this is correct as it stands.  `ceph auth` adds the keyring to the mons.

File: doc/rados/operations/control.rst [hunk 2]
```diff
@@ -106,22 +106,27 @@ to be considered ``stuck`` (default: 300).
...
+
     PGs are in an unknown state, because the OSDs that host them have not
     reported to the monitor cluster for a certain period of time (specified by
     the ``mon_osd_report_timeout`` configuration setting).
 
 
-To delete a ``lost`` RADOS object or revert an object to its prior state
-(either by reverting it to its previous version or by deleting it because it
-was just created and has no previous version), run the following command:
+To delete a ``lost`` object or revert an object to its prior state (either by
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:127]: Suggest factoring the parenthetical into a second sentence.

File: doc/rados/operations/control.rst [hunk 3]
```diff
@@ -169,9 +174,8 @@ To dump the OSD map, run the following command:
 
 The ``--format`` option accepts the following arguments: ``plain`` (default),
 ``json``, ``json-pretty``, ``xml``, and ``xml-pretty``. As noted above, JSON
-format is the recommended format for consumption by tools, scripting, and other
-forms of automation. 
-
+format is the recommended format for tools, scripting,
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:177]: elide the first `format`

File: doc/rados/operations/control.rst [hunk 4]
```diff
@@ -181,15 +185,15 @@ command:
 
    ceph osd tree [--format {format}]
 
-To find out where a specific RADOS object is stored in the system, run a
-command of the following form:
+To find out where a specific object is stored in the system, run a command of
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:188]: Why take out RADOS?

File: doc/rados/operations/control.rst [hunk 5]
```diff
@@ -181,15 +185,15 @@ command:
...
+the following form:
 
 .. prompt:: bash $
 
    ceph osd map <pool-name> <object-name>
 
 To add or move a new OSD (specified by its ID, name, or weight) to a specific
-CRUSH location, run the following command:
+location, run the following command:
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:196]: I think we want CRUSH here

File: doc/rados/operations/control.rst [hunk 6]
```diff
@@ -214,8 +218,8 @@ run the following command:
 
    ceph osd crush move {id} {loc1} [{loc2} ...]
 
-To set the CRUSH weight of a specific OSD (specified by ``{name}``) to
-``{weight}``, run the following command:
+To set the weight of a specific OSD (specified by ``{name}``) to ``{weight}``,
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:221]: And here, to disambiguate vs the override reweight

File: doc/rados/operations/control.rst [hunk 7]
```diff
@@ -287,37 +291,37 @@ following command:
 
    ceph osd in {osd-num}
 
-By using the ``pause`` and ``unpause`` flags in the OSD map, you can pause or
-unpause I/O requests. If the flags are set, then no I/O requests will be sent
-to any OSD. If the flags are cleared, then pending I/O requests will be resent.
-To set or clear these flags, run one of the following commands:
+By using the "pause flags" in the OSD map, you can pause or unpause I/O
+requests.  If the flags are set, then no I/O requests will be sent to any OSD.
+If the flags are cleared, then pending I/O requests will be resent. To set or
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:296]: s/if/when/

File: doc/rados/operations/control.rst [hunk 8]
```diff
@@ -287,37 +291,37 @@ following command:
...
+the same weight will receive approximately the same number of I/O requests and
+store approximately the same amount of data. The ``ceph osd reweight`` command
+assigns an override weight to an OSD. The weight value is in the range 0 to 1,
+and the command forces CRUSH to relocate a certain amount (1 - ``weight``) of
+the data that would otherwise be on this OSD. The command does not change the
+weights of the buckets above the OSD in the CRUSH map. Using the command is
+merely a corrective measure: for example, if one of your OSDs is at 90% and the
+others are at 50%, you could reduce the outlier weight to correct this
+imbalance. To assign an override weight to a specific OSD, run the following
+command:
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:316]: Add a note that override reweights conflict with the balancer, so if the balancer is in use, all override reweight values should be 1.0000 or behavior will be ... not what one wants.

File: doc/rados/operations/control.rst [hunk 9]
```diff
@@ -287,37 +291,37 @@ following command:
...
 
 .. prompt:: bash $
 
    ceph osd reweight {osd-num} {weight}
 
 A cluster's OSDs can be reweighted in order to maintain balance if some OSDs
 are being disproportionately utilized. Note that override or ``reweight``
-weights have relative values that default to 1.00000. Their values are not
+weights have relative values that default to 1.00000; their values are not
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:324]: Maybe clarify that they're relative to *each other*

File: doc/rados/operations/control.rst [hunk 10]
```diff
@@ -351,17 +355,9 @@ can be useful in certain circumstances: for example, when you are hastily
...
-.. note:: The ``balancer`` module does the work for you and achieves a more
-   uniform result, shuffling less data along the way. When enabling the
-   ``balancer`` module, you will want to converge any changed override weights
-   back to 1.00000 so that the balancer can do an optimal job. If your cluster
-   is very full, reverting these override weights before enabling the balancer
-   may cause some OSDs to become full. This means that a phased approach may
-   needed.
+Operators of deployments that utilize Nautilus (or later revisions of Luminous
+and Mimic) and that have no pre-Luminous clients might instead want to enable
+the `balancer`` module for ``ceph-mgr``.
```
Comment (anthonyeleven) [inline doc/rados/operations/control.rst:360]: I think I wrote this several releases ago, and the balancer is on by default now?

**Review by anthonyeleven — APPROVED** [review]

File: doc/rados/operations/control.rst (hunk 1 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:63]: Good catch. Thanks.

Accepted.

File: doc/rados/operations/control.rst (hunk 2 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:127]: I've just removed the parentheses, and I think the sentence works now.

File: doc/rados/operations/control.rst (hunk 3 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:177]: Accepted.

File: doc/rados/operations/control.rst (hunk 4 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:188]: Fair enough. It's back in.

File: doc/rados/operations/control.rst (hunk 5 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:196]: It has been added back in.

File: doc/rados/operations/control.rst (hunk 6 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:221]: It has also been added back in.

File: doc/rados/operations/control.rst (hunk 7 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:296]: Accepted.

File: doc/rados/operations/control.rst (hunk 8 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:316]: I have added a note.

File: doc/rados/operations/control.rst (hunk 10 above)
Comment (zdover23) [reply doc/rados/operations/control.rst:360]: I will ask later tonight, and return in what I hope is a near-future PR to address this matter.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53811 - Reef backport
https://github.com/ceph/ceph/pull/53812 - Quincy backport

**Pull request merged**